### PR TITLE
feat: feedback on pdf from provided url

### LINKED ISSUE #47: Allow bot to feedback on the PDF url given
- Research ways to get the bot to gain access to the PDF. One way we decided is to put the PDF download link in the command since this is simple and straightforward to begin with. So you can follow this direction and find a tutorial that helps you do this.
- Implement in `src/discord_bot/`:
  - Download PDF
  - Parse PDF
  - Delete PDF
  - Send pdf parsed text to LLM
  - Handle errors

### PR DESCRIPTION
**Pull request by hungdtrn**
This resolves #47

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -92,7 +94,7 @@ warn_unused_ignores = true
 
 # Prevent mypy from trigger the missing import error for our written packages
 [[tool.mypy.overrides]]
-module = ["discord_bot.*", "src.discord_bot.*"]
+module = ["discord_bot.*", "fitz.*", "gdown.*"]
```
Comment (samhwang) [inline pyproject.toml:97]: Is this needed?

Comment (rodonguyen) [reply pyproject.toml:97]: yes, as explained in Discord

File: src/discord_bot/llm.py [hunk 2]
```diff
@@ -4,22 +4,76 @@
...
+    with fitz.open(pdf_path) as pdf:
+      text_list = []
+      for page in pdf:
+        text_list.append(page.get_text())
+      text = "\n\n".join(text_list)
+
+    return text
+
+  except Exception as e:
+    raise RuntimeError(f"Error in parsing PDF: {e}")
```
Comment (samhwang) [inline src/discord_bot/llm.py:73]: pdf related utilities should be in a different module. This `llm.py` module should only contain LLM related things.

File: src/discord_bot/bot.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 
 MODEL_CHOICES = ["gpt-3.5-turbo", "gpt-4", "phi"]
-DEFAULT_MODEL = MODEL_CHOICES[2]
+DEFAULT_MODEL = MODEL_CHOICES[1]
```
Comment (samhwang) [inline src/discord_bot/bot.py:8]: If we have a default model to run review resume with, does it make sense for the ask llm question command to also have a default model?

Comment (rodonguyen) [reply src/discord_bot/bot.py:8]: this seems to be a question for @hungdtrn and @nqngo.

Previously we wanted to offer users multiple models to choose from I think.

**Review by samhwang — CHANGES_REQUESTED** [review]
missing unit tests

**Review by samhwang — APPROVED** [review]

**Review by rodonguyen — APPROVED** [review]
Ohyeah

**Pull request merged**